src/docfx/lib/path/PackageUrl.cs

```diff
@@ -14,23 +15,26 @@ namespace Microsoft.Docs.Build
     /// The commit-sh can be any tag, sha, or branch. The default commit-ish is master.
     /// </summary>
     [JsonConverter(typeof(ShortHandConverter))]
-    internal readonly struct PackageUrl
+    internal class PackageUrl : IEquatable<PackageUrl>, IComparable<PackageUrl>
```

Comment:
Consider changing `RestoreGitMap.DependencyGitLock` to use `(url, branch)` as key because `PackageUrl` represents more than git and it does not need to implement IEquatable or IComparable

Reply:
RestoreGitMap can also get local path right now, so Package Url is perfect for current situation

Reply:
I will do it in another PR.